### ISSUE TITLE
building openexr/3.2.1 on the windows, failed with a cmake error.

### ISSUE THREAD

**Issue by 11janson** [conan-io/docs#3594]
opencv's dependencies contain openexr/3.2.1, so when I want to build opencv on the windows, failed with a openexr build error about cmake. 
______________________
this is the conanfile.py
`from conan import ConanFile

#conan install . --output-folder=./conanoutput --build=missing --profile=default_debug 

class CompressorRecipe(ConanFile):
    settings = "os", "compiler", "build_type", "arch"
    generators = "CMakeToolchain", "CMakeDeps"
    options = {"shared":[True, False]}
    #default_options = {"shared": False}#ios下生成静态库
    default_options = {"shared": True}#windows下生成动态库

    def requirements(self):
        self.requires("pcl/1.13.1")
        self.requires("sophus/1.22.10")
        self.requires("laszip/3.4.3")
        self.requires("onetbb/2021.10.0")
       self.requires("opencv/4.8.1")
        self.requires("yaml-cpp/0.8.0")
        self.requires("ceres-solver/2.2.0")
        self.requires("libpcap/1.10.4")

        self.requires("libpng/1.6.42", override=True)#解决版本冲突问题
        self.requires("libdeflate/1.19")

       def build_requirements(self):
        # use the system's CMake for Windows
        if self.settings.os != "Windows":
            self.tool_requires("cmake/3.22.6")

`
and the logs:

-------- Installing package openexr/3.2.1 (44 of 60) --------
openexr/3.2.1: Building from source
openexr/3.2.1: Package openexr/3.2.1:0c081c6cd866b2575075a54d57174be9ff11ca53
openexr/3.2.1: Copying sources to build folder
openexr/3.2.1: Building your package in C:\Users\janson.yang.FJZ\.conan2\p\b\openebb015446044ca\b\build\generators
openexr/3.2.1: CMakeToolchain generated: conan_toolchain.cmake
openexr/3.2.1: CMakeToolchain generated: CMakePresets.json
openexr/3.2.1: CMakeToolchain generated: ..\..\src\CMakeUserPresets.json
openexr/3.2.1: Generating aggregated env files
openexr/3.2.1: Generated aggregated env files: ['conanbuild.bat', 'conanrun.bat']
openexr/3.2.1: Calling build()
openexr/3.2.1: Apply patch (portability): Workaround for GCC 5 bug
openexr/3.2.1: Running CMake.configure()
openexr/3.2.1: RUN: cmake -G "Visual Studio 17 2022" -DCMAKE_TOOLCHAIN_FILE="C:/Users/janson.yang.FJZ/.conan2/p/b/openebb015446044ca/b/build/generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="C:/Users/janson.yang.FJZ/.conan2/p/b/openebb015446044ca/p" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" "C:\Users\janson.yang.FJZ\.conan2\p\b\openebb015446044ca\b\src"
-- Using Conan toolchain: C:/Users/janson.yang.FJZ/.conan2/p/b/openebb015446044ca/b/build/generators/conan_toolchain.cmake
-- Conan toolchain: CMAKE_GENERATOR_TOOLSET=v143
-- Conan toolchain: C++ Standard 17 with extensions OFF
-- Conan toolchain: Setting BUILD_SHARED_LIBS = OFF
-- Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.19044.
-- The C compiler identification is MSVC 19.38.33130.0
-- The CXX compiler identification is MSVC 19.38.33130.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.38.33130/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.38.33130/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - not found
-- Found Threads: TRUE  
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- libdeflate was not found, installing from https://github.com/ebiggers/libdeflate.git (v1.18)
适用于 .NET Framework MSBuild 版本 17.8.3+195e7f5a3

Checking Build System
  Creating directories for 'deflate-populate'
  Building Custom Rule C:/Users/janson.yang.FJZ/.conan2/p/b/openebb015446044ca/b/build/_deps/deflate-subbuild/CMakeLists.txt
  Performing download step (git clone) for 'deflate-populate'
  Cloning into 'deflate-src'...
  fatal:unable to access 'https://github.com/ebiggers/libdeflate.git/': Failed to connect to github.com port 443: Bad access
Cloning into 'deflate-src'...
  fatal: unable to access 'https://github.com/ebiggers/libdeflate.git/': Failed to connect to github.com port 443: Bad access
  Cloning into 'deflate-src'...
  fatal: unable to access 'https://github.com/ebiggers/libdeflate.git/': Failed to connect to github.com port 443: Bad access
  -- Had to git clone more than once: 3 times.
  CMake Error at deflate-subbuild/deflate-populate-prefix/tmp/deflate-populate-gitclone.cmake:39 (message):
    Failed to clone repository: 'https://github.com/ebiggers/libdeflate.git'

CMake Error at C:/Program Files/CMake/share/cmake-3.23/Modules/FetchContent.cmake:1088 (message):
  Build step for deflate failed: 1
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.23/Modules/FetchContent.cmake:1217:EVAL:2 (__FetchContent_directPopulate)
  C:/Program Files/CMake/share/cmake-3.23/Modules/FetchContent.cmake:1217 (cmake_language)
  cmake/OpenEXRSetup.cmake:186 (FetchContent_Populate)
  CMakeLists.txt:56 (include)

-- Configuring incomplete, errors occurred!
See also "C:/Users/janson.yang.FJZ/.conan2/p/b/openebb015446044ca/b/build/CMakeFiles/CMakeOutput.log".
See also "C:/Users/janson.yang.FJZ/.conan2/p/b/openebb015446044ca/b/build/CMakeFiles/CMakeError.log".

openexr/3.2.1: ERROR:
Package '0c081c6cd866b2575075a54d57174be9ff11ca53' build failed
openexr/3.2.1: WARN: Build folder C:\Users\janson.yang.FJZ\.conan2\p\b\openebb015446044ca\b\build
ERROR: openexr/3.2.1: Error in build() method, line 72
        cmake.configure()
        ConanException: Error 1 while executing

**Comment by memsharded** [comment]
The error message says:

```
Cloning into 'deflate-src'...
fatal: unable to access 'https://github.com/ebiggers/libdeflate.git/': Failed to connect to github.com port 443: Bad access
```

Are you sure you have access to that URL: https://github.com/ebiggers/libdeflate/
Can you ``git clone https://github.com/ebiggers/libdeflate/`` it without problems in that machine?

**Comment by 11janson** [comment]
> The error message says:
> 
> ```
> Cloning into 'deflate-src'...
> fatal: unable to access 'https://github.com/ebiggers/libdeflate.git/': Failed to connect to github.com port 443: Bad access
> ```
> 
> Are you sure you have access to that URL: https://github.com/ebiggers/libdeflate/ Can you `git clone https://github.com/ebiggers/libdeflate/` it without problems in that machine?

yes, you are right. Need to set some git proxy, now I build openexr successfully. Thank you.
